src/box/vy_scheduler.c

```diff
@@ -1352,7 +1352,17 @@ vy_task_dump_abort(struct vy_task *task)
 	error_log(e);
 	say_error("%s: dump failed", vy_lsm_name(lsm));
 
-	vy_run_discard(task->new_run);
+	/*
+	 * The LSM tree could have been dropped while we were writing the new
+	 * run. In this case we should discard the run without committing to
+	 * vylog, because all the information about the LSM tree and its runs
+	 * could have already been garbage collected from vylog.
+	 */
+	if (lsm->is_dropped) {
+		vy_run_unref(task->new_run);
+	} else {
+		vy_run_discard(task->new_run);
+	}
```

Comment:
We don't embrace single-statement ifs [usually](https://www.tarantool.io/en/doc/latest/contributing/c_style_guide/#chapter-3-placing-braces-and-spaces) (also applied to the change below).

Reply:
Fixed, thanks!